docs/csharp/proto2.md

```diff
@@ -0,0 +1,184 @@
+Proto2 support in Google.Protobuf is finally here! This document outlines the new changes brought about 
+by this initial release of proto2 support. The generated code and public API associated with proto2 
+is experimental and subject to change in the future. APIs may be added, removed, or adjusted as feedback is received.
```

Comment:
optional nit: change to "APIs as well as the generated code may ..."? 
internal note: of course even though proto2 has been declared experimental, we can only change the apis/ generated code so that it doesn't break existing proto3 users (who are relying on the API stability and the proto2 support being there is irrelevant to them).